test/e2e/nodeportlocal_test.go

```diff
@@ -167,12 +169,30 @@ func updateNPLPortRangeInConfigmap(t *testing.T, data *TestData, newStartPort, n
 	}
 }
 
-func validatePortInRange(t *testing.T, nplAnnotations []k8s.NPLAnnotation, start, end int) {
+func validatePortsInAnnotation(t *testing.T, r *require.Assertions, nplAnnotations []k8s.NPLAnnotation, start, end int, targetPorts sets.Int) {
+	require.Equal(t, len(targetPorts), len(nplAnnotations))
 	for i := range nplAnnotations {
-		if nplAnnotations[i].NodePort > end || nplAnnotations[i].NodePort < start {
-			t.Fatalf("Node port %d not in range: %d - %d", nplAnnotations[i].NodePort, start, end)
+		podPort := nplAnnotations[i].PodPort
+		r.True(targetPorts.Has(podPort), "Port %d in Pod annotation not found in set target ports :%v from Services", podPort, targetPorts)
+		targetPorts.Delete(podPort)
+
+		nodePort := nplAnnotations[i].NodePort
+		if nodePort > end || nodePort < start {
+			t.Fatalf("Node port %d not in range: %d - %d", nodePort, start, end)
 		}
 	}
+	r.Emptyf(targetPorts, "Target ports %v not found in Pod annotation", targetPorts)
+}
+
+// createNginxPodWithPort creates an nginx Pod in the test namespace with configurable container port.
+func (data *TestData) createNginxPodWithPort(name, nodeName string, port int32) error {
+	return data.createPodOnNode(name, nodeName, nginxImage, []string{}, nil, nil, []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: port,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}, false, nil)
 }
```

Comment:
I am still unsure about this function and it's use: since NPL doesn't require the ContainerPort to be set anymore, what's the point?

Also it doesn't seem that there is any value in setting it to 90 in tests. I feel like it is actually misleading since we never use 90 as the Service targetPort. Am I missing something?

Reply:
I thought that adding a different port number in the pod explicitly would show that NPL doesn't depend on that port. But yeah, we don't need that. Removed.